=================================== FAILURES ===================================
  _________________________ test_click7_autocomplete_arg _________________________
  
      @pytest.mark.skipif(
          hasattr(click, "shell_complete"),
          reason="click-v7 old autocomplete function is not available, so skipped",
      )
      def test_click7_autocomplete_arg():
          def shell_complete_func(ctx, args, incomplete):
              return [name for name in ("foo", "bar") if name.startswith(incomplete)]
      
          @root_command.command()
          @click.argument("handler", autocompletion=shell_complete_func)
  >       def autocompletion_arg_cmd2(handler):
  
  tests/test_completion/test_click_version_le_7/test_arg_completion_v7.py:25:

--- a/click_repl/_repl.py
+++ b/click_repl/_repl.py
@@ -11,7 +11,11 @@ from .core import ReplContext
 from .exceptions import ClickExit  # type: ignore[attr-defined]
 from .exceptions import CommandLineParserError, ExitReplException, InvalidGroupFormat
 from .globals_ import ISATTY, get_current_repl_ctx
-from .utils import _execute_internal_and_sys_cmds
+from .utils import (
+    _execute_internal_and_sys_cmds,
+    _get_protected_args,
+    _set_protected_args,
+)
 
 __all__ = ["bootstrap_prompt", "register_repl", "repl"]
 
@@ -147,12 +151,12 @@ def repl(
 
             try:
                 # The group command will dispatch based on args.
-                old_protected_args = group_ctx.protected_args
+                old_protected_args = _get_protected_args(group_ctx)
                 try:
-                    group_ctx.protected_args = args
+                    _set_protected_args(group_ctx, args)
                     group.invoke(group_ctx)
                 finally:
-                    group_ctx.protected_args = old_protected_args
+                    _set_protected_args(group_ctx, old_protected_args)
             except click.ClickException as e:
                 e.show()
             except (ClickExit, SystemExit):

--- a/click_repl/utils.py
+++ b/click_repl/utils.py
@@ -29,6 +29,20 @@ __all__ = [
 ]
 
 
+def _get_protected_args(ctx: click.Context) -> list[str]:
+    # In click >= 8.2, protected_args is a read-only property over _protected_args.
+    if hasattr(ctx, "_protected_args"):
+        return t.cast(list[str], getattr(ctx, "_protected_args"))
+    return ctx.protected_args
+
+
+def _set_protected_args(ctx: click.Context, args: list[str]) -> None:
+    if hasattr(ctx, "_protected_args"):
+        setattr(ctx, "_protected_args", args)
+    else:
+        ctx.protected_args = args
+
+
 def _resolve_context(args: list[str], ctx: click.Context) -> click.Context:
     """Produce the context hierarchy starting with the command and
     traversing the complete arguments. This only follows the commands,
@@ -49,7 +63,7 @@ def _resolve_context(args: list[str], ctx: click.Context) -> click.Context:
                     return ctx
 
                 ctx = cmd.make_context(name, args, parent=ctx, resilient_parsing=True)
-                args = ctx.protected_args + ctx.args
+                args = _get_protected_args(ctx) + ctx.args
             else:
                 while args:
                     name, cmd, args = command.resolve_command(ctx, args)
@@ -68,7 +82,7 @@ def _resolve_context(args: list[str], ctx: click.Context) -> click.Context:
                     args = sub_ctx.args
 
                 ctx = sub_ctx
-                args = [*sub_ctx.protected_args, *sub_ctx.args]
+                args = [*_get_protected_args(sub_ctx), *sub_ctx.args]
         else:
             break
 

--- a/setup.cfg
+++ b/setup.cfg
@@ -27,7 +27,7 @@ packages=
   click_repl
 
 install_requires =
-  click>=7.0,<8.2.0
+  click>=7.0,<9.0
   prompt_toolkit>=3.0.36
   typing-extensions>=4.7.0
 

--- a/setup.cfg
+++ b/setup.cfg
@@ -15,12 +15,12 @@ license = MIT
 classifiers =
     Programming Language :: Python :: 3
     Programming Language :: Python :: 3 :: Only
-    Programming Language :: Python :: 3.6
-    Programming Language :: Python :: 3.7
-    Programming Language :: Python :: 3.8
     Programming Language :: Python :: 3.9
     Programming Language :: Python :: 3.10
     Programming Language :: Python :: 3.11
+    Programming Language :: Python :: 3.12
+    Programming Language :: Python :: 3.13
+    Programming Language :: Python :: 3.14
 
 [options]
 packages=
@@ -31,7 +31,7 @@ install_requires =
   prompt_toolkit>=3.0.36
   typing-extensions>=4.7.0
 
-python_requires = >=3.6
+python_requires = >=3.9
 zip_safe = no
 
 [options.extras_require]

